src/main/java/net/dv8tion/jda/api/requests/restaction/GuildScheduledEventAction.java

```diff
@@ -0,0 +1,286 @@
+     * voice channel info will be lost!
+     * <p><b>Note:</b> A location is required to be set by either this method, {@link #setLocation(VoiceChannel)} or
+     * {@link #setLocation(String)} before the event is created.
+     *
+     * @param  stageChannel
+     *         The Stage Channel that the new {@link GuildScheduledEvent} is set to take place in.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided Stage Channel is {@code null}, or is not from the same guild
+     *         that the event is being created in.
+     *
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         If the currently logged in account does not have
+     *         {@link net.dv8tion.jda.api.Permission#MANAGE_EVENTS Permission.ManageEvents},
+     *         {@link net.dv8tion.jda.api.Permission#MANAGE_CHANNEL Permission.MANAGE_CHANNEL},
+     *         {@link net.dv8tion.jda.api.Permission#VOICE_MUTE_OTHERS Permission.VOICE_MUTE_OTHERS},
+     *         or {@link net.dv8tion.jda.api.Permission#VOICE_MOVE_OTHERS Permission.VOICE_MOVE_OTHERS}, in the provided
+     *         stage channel.
+     *
+     * @return The current GuildScheduledEventAction, for chaining convenience
+     */
+    @Nonnull
+    @CheckReturnValue
+    GuildScheduledEventAction setLocation(@Nonnull StageChannel stageChannel);
+
+    /**
+     * Sets the location for the new {@link GuildScheduledEvent} to take place in a specified voice channel. This will
+     * also change the event's type to {@link GuildScheduledEvent.Type#VOICE}, and any previously set external location
+     * or stage channel info will be lost!
+     * <p><b>Note:</b> A location is required to be set by either this method, {@link #setLocation(StageChannel)} or {@link #setLocation(String)} before the
+     * event is created.
+     *
+     * @param  voiceChannel
+     *         The Voice Channel that the selected {@link GuildScheduledEvent} is set to take place at.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided Voice Channel is {@code null}, or is not from the same guild
+     *         that the event is being created in.
+     *
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         If the currently logged in account does not have
+     *         {@link net.dv8tion.jda.api.Permission#MANAGE_EVENTS Permission.ManageEvents},
+     *         {@link net.dv8tion.jda.api.Permission#VIEW_CHANNEL Permission.VIEW_CHANNEL}, or
+     *         {@link net.dv8tion.jda.api.Permission#VOICE_CONNECT Permission.VOICE_CONNECT} in the provided
+     *         Voice Channel.
+     *
+     * @return The current GuildScheduledEventAction, for chaining convenience
+     */
+    @Nonnull
+    @CheckReturnValue
+    GuildScheduledEventAction setLocation(@Nonnull VoiceChannel voiceChannel);
+
```

Comment:
Could do that, but it does not really make anything better. It would add more code having to check the channel type and be less intuitive for the user.